nat-lab/tests/test_mesh_firewall.py

```diff
@@ -69,28 +106,100 @@ async def test_mesh_firewall_successful_passthrough() -> None:
             )
         )
 
-        async with Ping(connection_alpha, beta.ip_addresses[0]).run() as ping:
-            with pytest.raises(asyncio.TimeoutError):
+        if alpha_ip_stack in [IPStack.IPv4, IPStack.IPv4v6] and beta_ip_stack in [
+            IPStack.IPv4,
+            IPStack.IPv4v6,
+        ]:
+            async with Ping(
+                connection_alpha,
+                testing.unpack_optional(beta.get_ip_address(IPProto.IPv4)),
+            ).run() as ping:
+                with pytest.raises(asyncio.TimeoutError):
+                    await testing.wait_long(ping.wait_for_next_ping())
+
+            async with Ping(
+                connection_beta,
+                testing.unpack_optional(alpha.get_ip_address(IPProto.IPv4)),
+            ).run() as ping:
                 await testing.wait_long(ping.wait_for_next_ping())
 
-        async with Ping(connection_beta, alpha.ip_addresses[0]).run() as ping:
-            await testing.wait_long(ping.wait_for_next_ping())
+            # this should still block
+            async with Ping(
+                connection_alpha,
+                testing.unpack_optional(beta.get_ip_address(IPProto.IPv4)),
+            ).run() as ping:
+                with pytest.raises(asyncio.TimeoutError):
+                    await testing.wait_long(ping.wait_for_next_ping())
 
-        # this should still block
-        async with Ping(connection_alpha, beta.ip_addresses[0]).run() as ping:
-            with pytest.raises(asyncio.TimeoutError):
-                await testing.wait_long(ping.wait_for_next_ping())
+        if alpha_ip_stack in [IPStack.IPv6, IPStack.IPv4v6] and beta_ip_stack in [
+            IPStack.IPv6,
+            IPStack.IPv4v6,
+        ]:
+            async with Ping(
+                connection_alpha,
+                testing.unpack_optional(beta.get_ip_address(IPProto.IPv6)),
+            ).run() as ping6:
+                with pytest.raises(asyncio.TimeoutError):
+                    await testing.wait_long(ping6.wait_for_next_ping())
+
+            async with Ping(
+                connection_beta,
+                testing.unpack_optional(alpha.get_ip_address(IPProto.IPv6)),
+            ).run() as ping6:
+                await testing.wait_long(ping6.wait_for_next_ping())
+
+            # this should still block
+            async with Ping(
```

Comment:
This seems exactly the same as https://github.com/NordSecurity/libtelio/pull/96/files#diff-8e08000ac20488cac787ea3d92cc50ae6534dd4aadaf2c2496bd95ffa9b5221eR138 Probably some parameters are mixed up

Reply:
I just realized that this is a regresion test, where we had an issue long time ago, that after pinging from one side, one would be able to ping back.